runtime/common/src/test/kotlin/kotlinx/serialization/json/polymorphic/JsonMapPolymorphismTest.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json.polymorphic
+
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import kotlin.test.*
+
+class JsonMapPolymorphismTest : JsonTestBase() {
+
+    @BeforeTest
+    fun setUp() {
+        unquoted.install(polymorphicTestModule)
+    }
+
+    @Serializable
+    private data class MapWrapper(val map: Map<String, @Polymorphic InnerBase>)
+
+    @Test
+    fun testPolymorphicValues() = parametrizedTest { useStreaming ->
+        val wrapper = MapWrapper(mapOf("k1" to InnerImpl(1), "k2" to InnerImpl2(2)))
+        val serialized = unquoted.stringify(wrapper, useStreaming)
+        assertEquals("{map:" +
+                "{k1:{\$type:kotlinx.serialization.json.polymorphic.InnerImpl,field:1,str:default,nullable:null}," +
+                "k2:{\$type:kotlinx.serialization.json.polymorphic.InnerImpl2,field:2}}}", serialized)
+        assertEquals(wrapper, unquoted.parse(serialized, useStreaming))
+    }
+
+    @Serializable
+    private data class MapNullableWrapper(val map: Map<String, @Polymorphic InnerBase?>)
+
+    @Test
+    fun testPolymorphicNullableValues() = parametrizedTest { useStreaming ->
+        val wrapper = MapNullableWrapper(mapOf("k1" to InnerImpl(1), "k2" to null))
```

Comment:
We can probably combine somehow `parameterizedTest` and `assertStringFormAndRestored` to avoid repeating these 4 lines with different arguments